src/DDTrace/Integrations/Nette/NetteIntegration.php

```diff
@@ -0,0 +1,176 @@
+                $span->name = 'nette.application.run';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+                $root->setTag(Tag::HTTP_STATUS_CODE, http_response_code());
+            }
+        );
+
+        \DDTrace\trace_method(
+            'Nette\Application\UI\Presenter',
+            'run',
+            function (SpanData $span, $args) use ($root, $service) {
+
+                $span->name = 'nette.presenter.run';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+
+                if (count($args) < 1 || !\is_a($args[0], '\Nette\Application\Request')) {
+                    return;
+                }
+
+                $request = $args[0];
+                $presenter = $request->getPresenterName();
+                $action = $request->getParameter('action');
+
+                $root->setTag(Tag::HTTP_METHOD, $request->getMethod());
+                $root->setTag('nette.route.presenter', $presenter);
+                $root->setTag('nette.route.action', $action);
+                $root->setResource($presenter . ':' . $action);
+            }
+        );
+
+        \DDTrace\trace_method(
+            'Nette\Application\Routers\RouteList',
+            'match',
+            function (SpanData $span) use ($service) {
+                $span->name = 'nette.router.match';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+            }
+        );
+
+        // Latte template engine traces
+        \DDTrace\trace_method(
+            'Latte\Engine',
+            'createTemplate',
+            function (SpanData $span, $args) use ($service) {
+                $span->name = 'nette.latte.createTemplate';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+
+                if (count($args) >= 1) {
+                    $span->meta['nette.latte.templateName'] = $args[0];
+                }
+            }
+        );
+
+        \DDTrace\trace_method(
+            'Latte\Engine',
+            'render',
+            function (SpanData $span, $args) use ($service) {
+                $span->name = 'nette.latte.render';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+
+                if (count($args) >= 1) {
+                    $span->meta['nette.latte.templateName'] = $args[0];
+                }
+            }
+        );
+
+        \DDTrace\trace_method(
+            'Latte\Engine',
+            'renderToString',
+            function (SpanData $span, $args) use ($service) {
+                $span->name = 'nette.latte.render';
+                $span->type = Type::WEB_SERVLET;
+                $span->service = $service;
+
+                if (count($args) >= 1) {
+                    $span->meta['nette.latte.templateName'] = $args[0];
+                }
+            }
+        );
+    }
```

Comment:
@SammyK At one point we talked about instrumenting components of frameworks like templating systems individually in case they get mixed and matched, which some small percentage of customers do, though I don't think `latte/latte` is at the top of that list. Is that something you still think we should do? If so, should we start now?

Reply:
> Is that something you still think we should do? If so, should we start now?

I think we should totally do this but I think this makes a lot more sense starting with Symfony components. I would like spend the time to put together a strategy to target Symfony components specifically and then expand from there. So, yes, but only after we have a clear plan of action. :)